src/main/java/com/amazon/sqs/javamessaging/AmazonSQSExtendedClient.java

```diff
@@ -354,6 +354,9 @@ public ReceiveMessageResult receiveMessage(ReceiveMessageRequest receiveMessageR
             Optional<String> largePayloadAttributeName = getReservedAttributeNameIfPresent(message.getMessageAttributes());
             if (largePayloadAttributeName.isPresent()) {
                 String largeMessagePointer = message.getBody();
+                largeMessagePointer = largeMessagePointer.contains("software.amazon.payloadoffloading.PayloadS3Pointer") ?
```

Comment:
String manipulation of JSON seems uncouth.  Anyway, beyond that I guess it doesn't matter much but:
1. this `contains` seems unnecessary since the replace will no-op if it's not present
2. `replaceFirst` is regex-based, so the raw/unescaped `.` are matching any character; perhaps just use `replace(CharSequence, CharSequence)`?